File: Clojure/pi_sequential.clj
;  Calculation of Pi using quadrature. Sequential algorithm.
;
;  Copyright © 2009–2011, 2013  Russel Winder

(load "output")

(defn summation [count delta]
  (loop [
          i count
          s 0.0
         ]
    (if (zero? i)
      s
      (let [x (* (- i 0.5) delta)]
        (recur (dec i) (+ s (/ 1.0 (+ 1.0 (* x x)))))))))

(let [
       n  100000000 ; 10 times fewer due to speed issues.
       delta  (/ 1.0 n)
       startTimeNanos  (System/nanoTime)
       pi  (* 4.0 delta (summation n delta))
       elapseTime  (/ (- (System/nanoTime) startTimeNanos) 1e9)
      ]
  (out "Sequential" pi n elapseTime))

File: C#/Pi_Sequential.cs
/*
 *  Calculation of π using quadrature realized with a basic sequential algorithm.
 *
 *  Copyright © 2009, 2011, 2015  Russel Winder
 */

public class Pi_CS_Sequential {

  public static void Main (string[] args) {
    const int n = 1000000000;
    const double delta = 1.0 / n;
    long startTimeHundredsOfNanos = System.DateTime.Now.Ticks;
    double sum = 0.0;
    for (int i = 1; i <= n; ++i) {
      double x = (i - 0.5) * delta;
      sum += 1.0 / (1.0 + x * x);
    }
    double pi = 4.0 * delta * sum;
    double elapseTime = (System.DateTime.Now.Ticks - startTimeHundredsOfNanos) / 1e7;
    Output.output("Sequential", pi, n, elapseTime);
  }

}
